src/shared/components/Button/Button.style.ts

```diff
@@ -6,29 +6,33 @@ import { colors, typography } from '../../theme'
 export type ButtonStyleProps = {
   variant?: 'primary' | 'secondary' | 'tertiary'
   full?: boolean
-  size?: 'regular' | 'small' | 'smaller'
+  size?: 'large' | 'medium' | 'small'
   disabled?: boolean
   hasText?: boolean
   clickable?: boolean
 }
 
 export type IconStyleProps = {
   disabled?: boolean
+  size?: 'large' | 'medium' | 'small'
+  hasText?: boolean
 }
 
-const colorsFromProps = ({ variant }: ButtonStyleProps) => {
+const colorsFromProps = ({ variant, hasText }: ButtonStyleProps) => {
   let styles
   switch (variant) {
     case 'tertiary': {
+      const circleRadius = hasText ? '0' : '100%'
       styles = css`
         background-color: transparent;
         border-color: transparent;
-        color: ${colors.blue[500]};
+        color: ${colors.white};
         &:hover {
-          color: ${colors.blue[300]};
+          background-color: ${colors.gray[800]};
+          border-radius: ${circleRadius};
         }
         &:active {
-          color: ${colors.blue[700]};
+          background-color: transparent;
```

Comment:
Those are the ones I've mentioned were getting updated in the design, take a look at Figma